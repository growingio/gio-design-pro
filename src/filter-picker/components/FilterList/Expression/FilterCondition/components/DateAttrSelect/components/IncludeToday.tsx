import React from 'react';
import { Moment } from 'moment';
import { DatePicker } from '@gio-design/components';

interface IncludeTodayProps {
  time: Moment;
  onChange: (v: any) => void;
  attrSelect: string;
}
function IncludeToday(props: IncludeTodayProps) {
  const { time, onChange } = props;

  const changeDate = (value: Moment | null) => {
    onChange(value);
  };
  return (
    <>
      <DatePicker value={time} showFooter onChange={changeDate} format="YYYY/MM/DD" />
    </>
  );
}

export default IncludeToday;
