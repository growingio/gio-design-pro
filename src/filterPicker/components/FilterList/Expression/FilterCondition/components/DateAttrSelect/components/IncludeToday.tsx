import React from 'react';
import { Moment } from 'moment';
import { DatePicker } from '@gio-design-new/components';

interface IncludeTodayProps {
  time: Moment;
  onChange: (v: any) => void;
  attrSelect: string;
}
function IncludeToday(props: IncludeTodayProps) {
  const { time, onChange } = props;

  const changeDate = (value: Moment | null) => {
    onChange(value.valueOf());
  };
  return (
    <>
      <DatePicker value={time} showFooter onChange={changeDate} format="YYYY/MM/DD" />
    </>
  );
}

export default IncludeToday;
