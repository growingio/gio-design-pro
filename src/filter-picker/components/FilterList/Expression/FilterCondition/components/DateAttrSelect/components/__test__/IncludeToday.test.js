import React from 'react';
import { mount } from 'enzyme';
import moment from 'moment';
import { act } from 'react-test-renderer';
import IncludeToday from '../IncludeToday';

describe('IncludeToday', () => {
  it('test', () => {
    const time = moment('2021-03-01');
    const onChange = jest.fn();
    const wrapper = mount(<IncludeToday time={time} onChange={onChange} />);
    expect(wrapper.render()).toMatchSnapshot();
    act(() => {
      wrapper.find('.gio-input__content').simulate('click');
    });
    act(() => {
      wrapper.find('.gio-date-picker-cell').at(2).simulate('click');
    });
    act(() => {
      wrapper.find('.gio-date-picker-footer button').at(1).simulate('click');
    });
    expect(onChange).toHaveBeenCalledTimes(1);
  });
});
