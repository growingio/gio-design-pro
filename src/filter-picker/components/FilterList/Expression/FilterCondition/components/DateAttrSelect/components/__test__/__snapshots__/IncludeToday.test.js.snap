// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncludeToday test 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "gio-date-picker-wrap",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "gio-input gio-input--middle",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "gio-input__content",
              "placeholder": "请输入…",
              "type": "text",
              "value": "2021/03/01",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "class": "gio-date-picker-wrapper",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "class": "gio-date-picker-wrapper",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "gio-input gio-input--middle",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "gio-input__content",
                "placeholder": "请输入…",
                "type": "text",
                "value": "2021/03/01",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
